### ISSUE TITLE
Support enums with type parameters

### ISSUE THREAD

**Issue by dtolnay** [joshleeb/varianteq#1]
```rust
#[macro_use]
extern crate varianteq;

#[derive(VariantEq)]
enum E<A: ToOwned, B = usize> {
    A(A),
    B(B),
}
```

I get this error in the generated code:

```
error[E0243]: wrong number of type arguments: expected at least 1, found 0
 --> tests/tests.rs:4:10
  |
4 | #[derive(VariantEq)]
  |          ^^^^^^^^^ expected at least 1 type argument
```

**Comment by dtolnay** [comment]
You would need to decide whether a generic enum can be PartialEq compared with the same enum but that has different generic parameters i.e. `E::A::<bool, u32>(true) == E::A::<bool, u64>(true)`.

**Comment by joshleeb** [comment]
I'll update to support enums. But I think that `E::A::<bool, u32>(true) != E::A::<bool, u64>(true)` since they are two different types.

**Comment by joshleeb** [comment]
Closed by #5
